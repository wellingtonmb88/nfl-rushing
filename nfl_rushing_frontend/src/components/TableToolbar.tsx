import React from 'react';
import clsx from 'clsx';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

import { EnhancedTableToolbarProps } from '../interfaces'
import { useToolbarStyles } from '../styles'

export default (props: EnhancedTableToolbarProps) => {
  const classes = useToolbarStyles();
  const { numSelected } = props;

  return (
    <Toolbar
      className={clsx(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
    >
      {numSelected > 0 ? (
        <Typography className={classes.title} color="inherit" variant="subtitle1">
          {numSelected} selected
        </Typography>
      ) : (
          <Typography className={classes.title} variant="h6" id="tableTitle">
            NFL Rushing Players
        </Typography>
        )}
    </Toolbar>
  );
};
