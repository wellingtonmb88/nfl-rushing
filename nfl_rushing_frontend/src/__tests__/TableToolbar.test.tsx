import React from "react";
import { create } from "react-test-renderer";
import TableToolbar from '../components/TableToolbar'

describe("TableToolbar component", () => {
  test("TableToolbar with numSelected > 0", () => {
    const component = create(<TableToolbar numSelected={1} />); 
    expect(component.toJSON()).toMatchSnapshot();
  });
  test("TableToolbar with numSelected == 0", () => {
    const component = create(<TableToolbar numSelected={0} />); 
    expect(component.toJSON()).toMatchSnapshot();
  });
});