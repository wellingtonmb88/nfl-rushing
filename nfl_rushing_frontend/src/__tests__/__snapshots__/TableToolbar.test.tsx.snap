// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolbar component TableToolbar with numSelected == 0 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular makeStyles-root-1 MuiToolbar-gutters"
>
  <h6
    className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
    id="tableTitle"
  >
    NFL Rushing Players
  </h6>
</div>
`;

exports[`TableToolbar component TableToolbar with numSelected > 0 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular makeStyles-root-1 makeStyles-highlight-2 MuiToolbar-gutters"
>
  <h6
    className="MuiTypography-root makeStyles-title-3 MuiTypography-subtitle1 MuiTypography-colorInherit"
  >
    1
     selected
  </h6>
</div>
`;
